File: src/com/tmrnk/gongon/dogage/module/PetAgeFragment.java
package com.tmrnk.gongon.dogage.module;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.tmrnk.gongon.dogage.R;
import com.tmrnk.gongon.dogage.activity.PetAgeActivity;
import com.tmrnk.gongon.dogage.entity.PetEntity;

/**
 * ペット年齢フラグメント
 * 
 * @access public
 */
public class PetAgeFragment extends Fragment
{
    private static final Integer[] BG_ABOVE = new Integer[] { R.drawable.img_frame_above1, R.drawable.img_frame_above2, R.drawable.img_frame_above3 };
    private static final Integer[] BG_UNDER = new Integer[] { R.drawable.img_frame_under1, R.drawable.img_frame_under2, R.drawable.img_frame_under3 };
    private static final Integer BG_CNT = 3;

    private PetEntity mItem = null;
    private Integer mPageNum = 0;
    private View mView = null;

    /**
     * onCreateView
     * 
     * @return void
     * @access public
     */
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        // fragment再生成抑止
        setRetainInstance(true);

        // データを取得
        Bundle bundle = getArguments();
        mItem = (PetEntity) bundle.getSerializable("item");
        mPageNum = bundle.getInt("pageNum");

        // レイアウトの指定
        mView = inflater.inflate(R.layout.fragment_pet_age, container, false);

        //viewPager初回表示の場合は、フラグメントを読み込んだタイミングでページを表示する
        PetAgeActivity activity = (PetAgeActivity) getActivity();
        activity.createInitPage();

        return mView;
    }

    /**
     * 画面表示
     * 
     * @return void
     * @access public
     */
    public void setDispItem()
    {
        TextView textViewName = (TextView) mView.findViewById(R.id.textViewName);
        textViewName.setText(mItem.getName());

        TextView textViewKind = (TextView) mView.findViewById(R.id.textViewKind);
        textViewKind.setText(mItem.getKindDisp());

        TextView textViewBirthday = (TextView) mView.findViewById(R.id.textViewBirthday);
        textViewBirthday.setText(mItem.getDispBirthday());

        TextView textViewPetAge = (TextView) mView.findViewById(R.id.textViewPetAge);
        textViewPetAge.setText(mItem.getPetAgeDisp());

        TextView textViewHumanAge = (TextView) mView.findViewById(R.id.textViewHumanAge);
        textViewHumanAge.setText(mItem.getHumanAge());

        TextView textViewDays = (TextView) mView.findViewById(R.id.textViewDays);
        textViewDays.setText(mItem.getDaysFromBorn());

        LinearLayout linearLayout1 = (LinearLayout) mView.findViewById(R.id.linearLayout1);
        linearLayout1.setBackgroundResource(BG_ABOVE[mPageNum % BG_CNT]);

        RelativeLayout relativeLayout1 = (RelativeLayout) mView.findViewById(R.id.relativeLayout1);
        relativeLayout1.setBackgroundResource(BG_UNDER[mPageNum % BG_CNT]);

        RelativeLayout relativeLayout2 = (RelativeLayout) mView.findViewById(R.id.relativeLayout2);
        relativeLayout2.setBackgroundResource(BG_UNDER[mPageNum % BG_CNT]);

        RelativeLayout relativeLayout3 = (RelativeLayout) mView.findViewById(R.id.relativeLayout3);
        relativeLayout3.setBackgroundResource(BG_UNDER[mPageNum % BG_CNT]);

    }
}

File: android/app/src/main/java/jp/co/e2/dogage/activity/PetAgeFragment.kt
package jp.co.e2.dogage.activity

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.fragment.app.Fragment
import com.squareup.picasso.Picasso
import jp.co.e2.dogage.R
import jp.co.e2.dogage.common.AndroidUtils
import jp.co.e2.dogage.config.Config
import jp.co.e2.dogage.dialog.PhotoDialog
import jp.co.e2.dogage.entity.PetEntity

/**
 * ペット年齢フラグメント
 */
class PetAgeFragment : Fragment() {

    companion object {
        val BG_ABOVE = arrayOf(R.drawable.img_frame_above1, R.drawable.img_frame_above2, R.drawable.img_frame_above3)
        val BG_UNDER = arrayOf(R.drawable.img_frame_under1, R.drawable.img_frame_under2, R.drawable.img_frame_under3)

        private const val PARAM_PAGE_NUM = "page_num"
        private const val PARAM_DATA = "data"

        /**
         * ファクトリーメソッド
         *
         * @return PetAgeFragment
         */
        fun newInstance(pageNum: Int, data: PetEntity): PetAgeFragment {
            val bundle = Bundle()
            bundle.putInt(PARAM_PAGE_NUM, pageNum)
            bundle.putSerializable(PARAM_DATA, data)

            val fragment = PetAgeFragment()
            fragment.arguments = bundle

            return fragment
        }
    }

    private val data: PetEntity by lazy { arguments!!.getSerializable(PARAM_DATA) as PetEntity }
    private val pageNum: Int by lazy { arguments!!.getInt(PARAM_PAGE_NUM) }

    /**
     * onCreateView
     *
     * @return void
     */
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreate(savedInstanceState)

        val view = inflater.inflate(R.layout.fragment_pet_age, container, false)
        
        // 画面表示
        setContent(view)

        return view
    }

    /**
     * 画面表示
     *
     * @param view
     */
    private fun setContent(view: View) {
        view.findViewById<TextView>(R.id.textViewName).apply {
            this.text = data.name
        }

        view.findViewById<TextView>(R.id.textViewKind).apply {
            this.text = data.getKindDisp(activity!!)
        }

        view.findViewById<TextView>(R.id.textViewBirthday).apply {
            this.text = data.getDispBirthday(activity!!)
        }

        view.findViewById<TextView>(R.id.textViewPetAge).apply {
            this.text = data.getPetAgeDisp(activity!!)
        }

        view.findViewById<TextView>(R.id.textViewHumanAge).apply {
            this.text = data.getHumanAge(activity!!)
        }

        view.findViewById<TextView>(R.id.textViewDays).apply {
            this.text = data.getDaysFromBorn(activity!!)
        }

        view.findViewById<LinearLayout>(R.id.linearLayout1).apply {
            this.setBackgroundResource(BG_ABOVE[pageNum % BG_ABOVE.size])
        }

        view.findViewById<ConstraintLayout>(R.id.constraintLayoutDogAge).apply {
            this.setBackgroundResource(BG_UNDER[pageNum % BG_ABOVE.size])
        }

        view.findViewById<ConstraintLayout>(R.id.constraintLayoutHumanAge).apply {
            this.setBackgroundResource(BG_UNDER[pageNum % BG_ABOVE.size])
        }

        view.findViewById<ConstraintLayout>(R.id.constraintLayoutDayCount).apply {
            this.setBackgroundResource(BG_UNDER[pageNum % BG_ABOVE.size])
        }

        view.findViewById<TextView>(R.id.textViewArchive).apply {
            if (data.archiveDate != null) {
                this.visibility = View.VISIBLE
                this.text = data.getDispArchiveDate(activity!!)
            } else {
                this.visibility = View.GONE
            }
        }

        view.findViewById<ImageView>(R.id.imageViewPhoto).apply {
            if (data.photoFlg) {
                val size = AndroidUtils.dpToPixel(activity, Config.PHOTO_THUMB_DP)
                Picasso.get().load(data.getImgFileUri(context)).resize(size, size).centerCrop().into(this)

                this.visibility = View.VISIBLE

                this.setOnClickListener {
                    val photoDialog = PhotoDialog.newInstance(data.getImgFileUri(context))
                    photoDialog.show(fragmentManager, "dialog")
                }
            } else {
                this.visibility = View.GONE
            }
        }
    }
}